twinkle.js

```diff
@@ -262,11 +262,24 @@ Twinkle.addPortlet = function(navigation, id, text, type, nextnodeid) {
 		}
 		return null;
 	}
+
 	if (type === 'menu') {
 		// In order to get mw.util.addPortlet to generate a dropdown menu in vector and vector-2022, the nextnodeid must be p-cactions. Any other nextnodeid will generate a non-dropdown portlet instead.
 		nextnodeid = 'p-cactions';
 	}
-	return mw.util.addPortlet(id, text, '#' + nextnodeid);
+
+	let portlet = mw.util.addPortlet(id, text, '#' + nextnodeid);
+
+	if (mw.config.get('skin') === 'vector') {
+		$('#p-twinkle').insertAfter('#p-cactions');
+	} else if (mw.config.get('skin') === 'vector-2022') {
+		$('#p-twinkle-dropdown').insertAfter('.vector-page-tools-landmark');
+		// this creates two #p-twinkle-dropdowns for some reason, in different spots on the DOM. maybe some JS in vector 2022? delete the bad one:
```

Comment:
I'm curious about why you are getting two p-twinkle-dropdowns  - could you provide some replication instructions? That doesn't seem expected.

Perhaps try detaching the node first before appending to avoid the need for this?

Reply:
Detach fixed it. Thank you for that tip!

Reply:
Just kidding, detach didn't fix it. Hmm.

Reply:
OK, fixed it by switching `insertAfter()` to `appendTo()`